internal/app/networkd/pkg/address/kernel.go

```diff
@@ -0,0 +1,140 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package address
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/talos-systems/talos/internal/pkg/kernel"
+	"golang.org/x/sys/unix"
+)
+
+// Kernel implements the Addressing interface
+type Kernel struct {
+	Address           string
+	ServerAddress     string
+	Gateway           string
+	Netmask           string
+	Hostname          string
+	Device            string
+	Auto              string
+	PrimaryResolver   string
+	SecondaryResolver string
+	NTPServer         string
+}
+
+// Discover doesnt do anything in the static configuration since all
+// the necessary configuration data is supplied via config.
+func (s *Kernel) Discover(ctx context.Context) error {
+	var option *string
+	if option = kernel.ProcCmdline().Get("ip").First(); option == nil {
+		return fmt.Error("no kernel.ip argument supplied, skipping")
+	}
+
+	if option == "dhcp" {
+		return fmt.Errorf("unsupported kernel.ip configuration method: %s", option)
+	}
+
+	// https://www.kernel.org/doc/Documentation/filesystems/nfs/nfsroot.txt
+	// ip=<client-ip>:<server-ip>:<gw-ip>:<netmask>:<hostname>:<device>:<autoconf>:<dns0-ip>:<dns1-ip>:<ntp0-ip>
+	r := strings.NewReader(strings.Join(strings.Split(option, ":"), " "))
+	_, err := fmt.Fscan(r,
```

Comment:
`Fscan` will require that `ip` has the full set of options specified, otherwise we will get `EOF`.

Reply:
I guess it's not required, but we will need to check if err is `EOF`.